Add comparison operators

thehammer opened the pull request:
Addition of standard comparison operators: ==, <, >, <=, >=

Qqwy commented:
Thank you very much for your Pull Request!

 It looks really nice, but I am a bit hesitant to add this functionality. The problem with operators (especially so for the comparison operators) is that people expect that they work in guard clauses, which they unfortunately do not (nor is it possible to create guard-clause-safe macros/functions that work on maps because you cannot pattern match inside a guard clause).

thehammer commented:
Heh, I debated going that route on the initial pull request for the exact reasons you listed. Opted for consistency on the flags defaulting to true.

I've made the change you requested (and caught a regression I introduced with the hidden functions). Also updated to use the atom return values for comparisons.

Stay awesome,

Hammer

Qqwy commented:
This change is included in the published version `2.1.0` :heart_eyes: . Thank you, @thehammer !
